ReactSkia/views/common/RSkTextUtils.cpp

```diff
@@ -0,0 +1,120 @@
+    for(auto &fragment: attributedString.getFragments()) {
+        if((paraHeight < frameHeight) && (!fragment.textAttributes.textAlignVertical.empty())) {
+            if(!strcmp(fragment.textAttributes.textAlignVertical.c_str(),"center"))
+                return ((frameHeight/2) - (paraHeight/2));
+            else if(!strcmp(fragment.textAttributes.textAlignVertical.c_str(),"bottom"))
+                return (frameHeight - paraHeight);
+        }
+    }
+    return 0;
+}
+} //namespace
+
+namespace RSkTextUtils{
+void setText(std::shared_ptr<Paragraph>& paragraph,
+            std::shared_ptr<ParagraphBuilder>& builder,
+            LayoutMetrics layoutout,
+            Rect frame,
+            ParagraphProps props) {
+    int numberOfLines = 0;
+    std::vector<LineMetrics> metrics;
+    ParagraphStyle paraStyle;
+
+    paragraph->layout(layoutout.getContentFrame().size.width);
+    paragraph->getLineMetrics(metrics);
+    numberOfLines = getTextLines(metrics,
+                                props.paragraphAttributes.maximumNumberOfLines,
+                                ((layoutout.contentInsets.top) || (layoutout.contentInsets.bottom)),
+                                frame.size.height,
+                                layoutout.getContentFrame().size.height);
+    if (props.paragraphAttributes.maximumNumberOfLines) {
+        if ((EllipsizeMode::Tail) == (props.paragraphAttributes.ellipsizeMode))
+            paraStyle.setEllipsis(u"\u2026");
+    }
+    
+    paraStyle.setMaxLines(numberOfLines);
+    builder->setParagraphStyle(paraStyle);
+    if (numberOfLines) {
+        paragraph = builder->Build();
+        paragraph->layout(layoutout.getContentFrame().size.width);
+    }
+
+    //return paragraph;
+}
+
+void drawText(std::shared_ptr<Paragraph>& paragraph,
+            AttributedString attributedString,
+            SkCanvas *canvas,
+            LayoutMetrics layout,
+            Rect frame,
+            ParagraphProps props) {
+    SkScalar yOffset = 0;
+    SkPaint paint;
+    auto fontLineHeight = (!std::isnan(props.textAttributes.lineHeight)) && (props.textAttributes.lineHeight >= 0) ?
+                                props.textAttributes.lineHeight :
+                                true;
+    auto fontOpacity = (!std::isnan(props.opacity)) && (props.opacity >= 0) && (props.opacity <= 1) ?
+                                props.opacity :
+                                true;
+
+    if ((props.backgroundColor) || (props.textAttributes.lineHeight >= 0)) {
+        paint.setColor(RSkColorFromSharedColor(props.backgroundColor, SK_ColorTRANSPARENT));
+        paint.setAntiAlias(true);
+        canvas->drawRect(SkRect::MakeXYWH(frame.origin.x, frame.origin.y, frame.size.width, frame.size.height), paint);
+        canvas->clipRect(SkRect::MakeXYWH(frame.origin.x,frame.origin.y,frame.size.width,frame.size.height));
+    }
+    if ((props.textAttributes.lineHeight) && (props.opacity)) {
```

Comment:
if layer is going to handle the opacity, is this check required here ?
@shefayaparwin @munezbn My understanding is that component will draw on canvas and keep. Layer will handle the opacity... and also if opacity changes, component dont need to redraw again ...

Reply:
Opacity will be handled in Layer paint. Component need not to handle